src/analysis/calculus/cont_diff.lean

```diff
@@ -2296,6 +2296,11 @@ lemma cont_diff.continuous_deriv (h : cont_diff 𝕜 n f₂) (hn : 1 ≤ n) :
   continuous (deriv f₂) :=
 (cont_diff_succ_iff_deriv.mp (h.of_le hn)).2.continuous
 
+lemma cont_diff.iterate_deriv :
+∀ (n : ℕ) (f₂ : 𝕜 → F) (hf : cont_diff 𝕜 ∞ f₂), cont_diff 𝕜 ∞ (deriv^[n] f₂)
```

Comment:
This is true more generally as something like
```suggestion
∀ (m n : ℕ) (hmn : n ≤ m) (f₂ : 𝕜 → F) (hf : cont_diff 𝕜 m f₂), cont_diff 𝕜 (m - n) (deriv^[n] f₂)
```
right? It would be nice to have both versions

Reply:
Yes, though personally I generally try to avoid using `nat.sub`. A subtraction-free phrasing is:
```
lemma cont_diff.iterate_deriv' (n : ℕ) : 
∀ (k : ℕ) (f₂ : 𝕜 → F) (hf : cont_diff 𝕜 (n + k : ℕ) f₂), cont_diff 𝕜 n (deriv^[k] f₂)
| 0       f₂ hf := hf
| (n + 1) f₂ hf := cont_diff.iterate_deriv'' n _ (cont_diff_succ_iff_deriv.mp hf).2
```
Here is a version phrased with `m-n` based on the above:
```
lemma cont_diff.iterate_deriv' (m n : ℕ) (hmn : n ≤ m) (f₂ : 𝕜 → F) (hf : cont_diff 𝕜 m f₂) :
  cont_diff 𝕜 (m - n) (deriv^[n] f₂) :=
begin
  obtain ⟨k, rfl⟩ := nat.exists_eq_add_of_le hmn,
  rw [← enat.coe_sub, nat.add_sub_cancel_left],
  rw add_comm at hf,
  clear hmn,
  induction n with n ih generalizing f₂,
  exact hf,
  rw nat.add_succ at hf,
  exact ih _ (cont_diff_succ_iff_deriv.mp hf).2,
end
```
(I also tried out doing the `nat.sub` approach directly with induction, and quickly fell into familiar `nat.sub` messiness.)